hypothesis-python/src/hypothesis/internal/conjecture/utils.py

```diff
@@ -172,14 +172,28 @@ def sample(self, data: "ConjectureData", forced: Optional[int] = None) -> int:
         forced_choice = (  # pragma: no branch # https://github.com/nedbat/coveragepy/issues/1617
             None
             if forced is None
-            else next((b, a, a_c) for (b, a, a_c) in self.table if forced in (b, a))
+            else next(
+                (b, a, a_c)
+                for (b, a, a_c) in self.table
+                if forced == b or (forced == a and a_c > 0)
```

Comment:
```suggestion
                (base, alternate, alternate_chance)
                for (base, alternate, alternate_chance) in self.table
                if forced == base or (forced == alternate and alternate_chance > 0)
```
nit: the short names only took a second to work out, but using the long ones removes even that hesitation.